### PR TITLE
add use strict and fix undeclared variables

### PR DESCRIPTION
**Pull request by Lebowskovitch**
I first had an issue with an `undeclared variable 'm'` using a request with a filter, then I noticed it was implicitly declared.
This works for some environments, but not in the browser. Mocha tests still pass because they don't run in strict mode, so the issue was silent.

I then added the `use strict` directive on test files and the `RequestConverter`, making the tests fail and report undeclared variables issues. 

tldr: this fixes undeclared variable issues and adds strict mode in some files.

### CODE REVIEW

**Comment by AleksandrRogov** [comment]
Great! Thanks!
